File: examples/src/Rendering-Printing/RenderOptions.cs
using Aspose.slides.Export
import aspose.slides as slides
import aspose.pydrawing as drawing
using System.Drawing.Imaging
using System.IO

namespace Aspose.slides.Examples.CSharp.Rendering.Printing
{
    # This example demonstrates one of the possible use cases of IRenderingOptions interface
    #(getting slide thumbnails with different default font and slide's notes shown)

    class RenderOptions
    {
        public static void Run()
        {
            # The path to the documents directory.
            dataDir = RunExamples.GetDataDir_Rendering()
            presPath = Path.Combine(dataDir, "RenderingOptions.pptx")

            using (Presentation pres = new Presentation(presPath))
            {
                IRenderingOptions renderingOpts = new RenderingOptions()
                renderingOpts.NotesCommentsLayouting.NotesPosition = NotesPositions.BottomTruncated

                pres.slides[0].GetThumbnail(renderingOpts, 4 / 3f, 4 / 3f).save(Path.Combine(RunExamples.OutPath, "RenderingOptions-Slide1-Original.png"), ImageFormat.Png)

                renderingOpts.NotesCommentsLayouting.NotesPosition = NotesPositions.NONE
                renderingOpts.DefaultRegularFont = "Arial Black"
                pres.slides[0].GetThumbnail(renderingOpts, 4 / 3f, 4 / 3f).save(Path.Combine(RunExamples.OutPath, "RenderingOptions-Slide1-ArialBlackDefault.png"), ImageFormat.Png)

                renderingOpts.DefaultRegularFont = "Arial Narrow"
                pres.slides[0].GetThumbnail(renderingOpts, 4 / 3f, 4 / 3f).save(Path.Combine(RunExamples.OutPath, "RenderingOptions-Slide1-ArialNarrowDefault.png"), ImageFormat.Png)
            }
        }
    }
}

File: examples/src/Rendering-Printing/RenderOptions.py
import aspose.slides as slides
import aspose.pydrawing as drawing

def rendering_options():
    # The path to the documents directory.
    dataDir = "./examples/data/"
    outDir = "./examples/out/"

    with slides.Presentation(dataDir + "rendering_options.pptx") as pres:
        renderingOpts = slides.export.RenderingOptions()
        renderingOpts.notes_comments_layouting.notes_position = slides.export.NotesPositions.BOTTOM_TRUNCATED

        pres.slides[0].get_thumbnail(renderingOpts, 4 / 3, 4 / 3).save(outDir + "rendering_options-Original.png", drawing.imaging.ImageFormat.png)

        renderingOpts.notes_comments_layouting.notes_position = slides.export.NotesPositions.NONE
        renderingOpts.default_regular_font = "Arial Black"
        pres.slides[0].get_thumbnail(renderingOpts, 4 / 3, 4 / 3).save(outDir + "rendering_options-ArialBlackDefault.png", drawing.imaging.ImageFormat.png)

        renderingOpts.default_regular_font = "Arial Narrow"
        pres.slides[0].get_thumbnail(renderingOpts, 4 / 3, 4 / 3).save(outDir + "rendering_options-ArialNarrowDefault.png", drawing.imaging.ImageFormat.png)



